From a1646032c65858c33904fa3c8135933446f4bc83 Mon Sep 17 00:00:00 2001
From: Rik van Riel <riel@redhat.com>
Date: Thu, 12 Mar 2015 17:47:00 -0400
Subject: [PATCH 2/3] fix cpulist_parse definition to match bitmap_parselist
 and kernel

Nobody used cpulist_parse before, so nobody noticed that the parameters
for cpulist_parse and bitmap_parselist did not match the parameters
actually used by bitmap_parselist and related functions.

Signed-off-by: Rik van Riel <riel@redhat.com>
---
 cpumask.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/cpumask.h b/cpumask.h
index 001dda3..0774a88 100644
--- a/cpumask.h
+++ b/cpumask.h
@@ -281,10 +281,10 @@ static inline int __cpulist_scnprintf(char *buf, int len,
 	return bitmap_scnlistprintf(buf, len, srcp->bits, nbits);
 }
 
-#define cpulist_parse(buf, dst) __cpulist_parse((buf), &(dst), NR_CPUS)
-static inline int __cpulist_parse(const char *buf, cpumask_t *dstp, int nbits)
+#define cpulist_parse(buf, len, dst) __cpulist_parse((buf), (len), &(dst), NR_CPUS)
+static inline int __cpulist_parse(const char *buf, int len, cpumask_t *dstp, int nbits)
 {
-	return bitmap_parselist(buf, dstp->bits, nbits);
+	return bitmap_parselist(buf, len, dstp->bits, nbits);
 }
 
 #define cpu_remap(oldbit, old, new) \
-- 
2.1.0

